### PR TITLE
[JXR-172] Upgrade Parent to 37, project cleanup

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- upgrade parent to 37
- remove configuration which override parent pom items
- use the same version of plugins in project and IT tests
- improvement assertions in IT tests

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/JXR) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[JXR-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `JXR-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: maven-jxr-plugin/src/it/settings.xml [hunk 1]
```diff
@@ -60,18 +46,10 @@ under the License.
...
-            <enabled>true</enabled>
-          </snapshots>
-        </pluginRepository>        
       </pluginRepositories>
     </profile>
   </profiles>
+  <activeProfiles>
+    <activeProfile>it-repo</activeProfile>
+  </activeProfiles>
 </settings>
```
Comment (michael-o) [inline maven-jxr-plugin/src/it/settings.xml:55]: Why do we need this settings file at all? I ran the tests without and they success. There is no settings file in MPIR.

**Review by michael-o — COMMENTED** [review]
If MPIR is used somewhere, please update to 3.4.0.

File: maven-jxr-plugin/src/it/settings.xml (hunk 1 above)
Comment (slawekjaranowski) [reply maven-jxr-plugin/src/it/settings.xml:55]: Needed for m-invoker-p https://maven.apache.org/plugins/maven-invoker-plugin/examples/fast-use.html

**Comment by slawekjaranowski** [comment]
> If MPIR is used somewhere, please update to 3.4.0.

Not used in IT test, project use version from parent 37 -> 3.3.0

File: maven-jxr-plugin/src/it/settings.xml (hunk 1 above)
Comment (michael-o) [reply maven-jxr-plugin/src/it/settings.xml:55]: Interesting, in other plugins I haven't seen it. Did you check the time difference actually?

Comment (michael-o) [reply maven-jxr-plugin/src/it/settings.xml:55]: Wow, right, but not used: https://github.com/apache/maven-project-info-reports-plugin/blob/a91b480cf52a152f1afbe52f691577aaf3f54866/pom.xml#L591-L611

Comment (slawekjaranowski) [reply maven-jxr-plugin/src/it/settings.xml:55]: One thing is time consumed be test and second is how many items are download from remote repository.

without `settings.xml`

```
[INFO] --- maven-invoker-plugin:3.3.0:run (integration-test) @ maven-jxr-plugin ---
[INFO] Building: simple-project/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           simple-project/pom.xml ........................... SUCCESS (23.1 s)
[INFO] Building: JXR-100_parameterlink/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-100_parameterlink/pom.xml .................... SUCCESS (3.5 s)
[INFO] Building: aggregate/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           aggregate/pom.xml ................................ SUCCESS (10.3 s)
[INFO] Building: mixed-plugin-report/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           mixed-plugin-report/pom.xml ...................... SUCCESS (13.0 s)
[INFO] Building: JXR-135_innerclasses/pom.xml
[INFO]           JXR-135_innerclasses/pom.xml ..................... SUCCESS (1.8 s)
[INFO] Building: JXR-143_nofork/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-143_nofork/pom.xml ........................... SUCCESS (5.5 s)
[INFO] Building: new-site-config/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           new-site-config/pom.xml .......................... SUCCESS (5.3 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 7, Failed: 0, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Maven JXR Parent 3.2.1-SNAPSHOT:
[INFO] 
[INFO] Maven JXR Parent ................................... SUCCESS [  2.545 s]
[INFO] Maven JXR .......................................... SUCCESS [  3.217 s]
[INFO] Maven JXR Plugin ................................... SUCCESS [01:10 min]
[INFO] ------------------------------------------------------------------------
```

with `settings.xml`

```
[INFO] --- maven-invoker-plugin:3.3.0:run (integration-test) @ maven-jxr-plugin ---
[INFO] Building: simple-project/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           simple-project/pom.xml ........................... SUCCESS (5.9 s)
[INFO] Building: JXR-100_parameterlink/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-100_parameterlink/pom.xml .................... SUCCESS (1.8 s)
[INFO] Building: aggregate/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           aggregate/pom.xml ................................ SUCCESS (6.1 s)
[INFO] Building: mixed-plugin-report/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           mixed-plugin-report/pom.xml ...................... SUCCESS (12.2 s)
[INFO] Building: JXR-135_innerclasses/pom.xml
[INFO]           JXR-135_innerclasses/pom.xml ..................... SUCCESS (1.6 s)
[INFO] Building: JXR-143_nofork/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           JXR-143_nofork/pom.xml ........................... SUCCESS (3.5 s)
[INFO] Building: new-site-config/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           new-site-config/pom.xml .......................... SUCCESS (4.2 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 7, Failed: 0, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Maven JXR Parent 3.2.1-SNAPSHOT:
[INFO] 
[INFO] Maven JXR Parent ................................... SUCCESS [  2.145 s]
[INFO] Maven JXR .......................................... SUCCESS [  2.473 s]
[INFO] Maven JXR Plugin ................................... SUCCESS [ 42.544 s]
[INFO] ------------------------------------------------------------------------
```

Comment (slawekjaranowski) [reply maven-jxr-plugin/src/it/settings.xml:55]: I will add do MIRP 😄

**Review by michael-o — APPROVED** [review]

File: maven-jxr-plugin/src/it/settings.xml (hunk 1 above)
Comment (slawekjaranowski) [reply maven-jxr-plugin/src/it/settings.xml:55]: hehe parent magic 😄 

https://github.com/apache/maven-parent/blob/550a810a356ca4479717771417cd9dea7b09ed8c/maven-plugins/pom.xml#L235

so MPIR use settings.xml in IT tests

Comment (michael-o) [reply maven-jxr-plugin/src/it/settings.xml:55]: Oh man, I was blind for years. Can we apply this here as well?

**Comment by jira-importer** [comment]
Resolve #284

**Pull request merged**